priam/src/main/java/com/netflix/priam/backup/IncrementalBackup.java

```diff
@@ -138,4 +140,19 @@ private Void deleteIfEmpty(File dir) {
         if (FileUtils.sizeOfDirectory(dir) == 0) FileUtils.deleteQuietly(dir);
         return null;
     }
+
+    public static int countFilesInBackupDir(IConfiguration config) throws Exception {
```

Comment:
My preference is that we not co-mingle this with classes dedicated to uploading incremental backups or taking snapshots. Please put these methods in backupV2Servlet or in a class with the purpose of counting the number of files. You'll need to share the constants INCREMENTAL_BACKUP_FOLDER and SNAPSHOT_FOLDER between all of them. The best place would be as static constants in BackupRestoreUtil.

Now that I think about it, the total bytes is really what is of interest. There is already an interface called DirectorySize and an implementation for Snapshot directories called SnapshotDirectorySize. You could implement a similar one for incrementals.

Reply:
Addressed this in new commit